### PR TITLE
Updates to handle llvm flag day change to CreateAtomicCmpXchg

### PR DESCRIPTION
**Pull request by dstutt**
Usual guards for old/new versions to be removed once everything has propagated.

### CODE REVIEW

File: lgc/patch/PatchBufferOp.cpp [hunk 1]
```diff
@@ -175,8 +175,16 @@ void PatchBufferOp::visitAtomicCmpXchgInst(AtomicCmpXchgInst &atomicCmpXchgInst)
...
     Value *const compareValue = atomicCmpXchgInst.getCompareOperand();
     Value *const newValue = atomicCmpXchgInst.getNewValOperand();
+#if LLVM_MAIN_REVISION && LLVM_MAIN_REVISION < 383129
+  // Old version of the code
     AtomicCmpXchgInst *const newAtomicCmpXchg =
         m_builder->CreateAtomicCmpXchg(atomicPointer, compareValue, newValue, successOrdering, failureOrdering);
+#else
+  // New version of the code (also handles unknown version, which we treat as latest)
+    AtomicCmpXchgInst *const newAtomicCmpXchg =
+        m_builder->CreateAtomicCmpXchg(atomicPointer, compareValue, newValue, MaybeAlign(),
```
Comment (dstutt) [inline lgc/patch/PatchBufferOp.cpp:185]: Do you think the MaybeAlign() here should be atomicCmpXchgInst.getAlign()? (and similarly for other cases where MaybeAlign() is used).
Note: Using MaybeAlign() gives the same functionality as before the change.

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp [hunk 2]
```diff
@@ -1778,7 +1778,13 @@ Value *SPIRVToLLVM::transAtomicRMW(SPIRVValue *const spvValue, const AtomicRMWIn
   Value *const atomicValue = transValue(spvAtomicInst->getOpValue(3), getBuilder()->GetInsertBlock()->getParent(),
                                         getBuilder()->GetInsertBlock());
 
+#if LLVM_MAIN_REVISION && LLVM_MAIN_REVISION < 383129
+  // Old version of the code
   return getBuilder()->CreateAtomicRMW(binOp, atomicPointer, atomicValue, ordering, scope);
+#else
+  // New version of the code (also handles unknown version, which we treat as latest)
+  return getBuilder()->CreateAtomicRMW(binOp, atomicPointer, atomicValue, MaybeAlign(), ordering, scope);
```
Comment (dstutt) [inline llpc/translator/lib/SPIRV/SPIRVReader.cpp:1786]: Similarly, is there a better way to generate the alignment argument here?

**Comment by amdvlk-admin** [comment]
### Test summary for commit d9bdb44c46084a9c4719b698a7aaf8a769445851
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev c9d1904b35ac5eb70bd4738a5bf5e537adbf5ab0</li>
<li>SPVGEN: dev 4846370016de9775b7518b51c08260387a7367c4</li>
<li>XGL: dev dd2834ff77dcb40511a18ae0ca678fafe8a339fe</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 28beb06b8e3555502626d39b9cba0892902886e1</li>
</ul></details>

<details><summary>CTS tests (Failed: 3/238435)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26198/59723 (43.9%)</li>
<li>Failed: 0/59723 (0.0%)</li>
<li>Skipped: 33525/59723 (56.1%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26198/59723 (43.9%)</li>
<li>Failed: 0/59723 (0.0%)</li>
<li>Skipped: 33525/59723 (56.1%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25456/59266 (43.0%)</li>
<li>Failed: 0/59266 (0.0%)</li>
<li>Skipped: 33810/59266 (57.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26858/59723 (45.0%)</li>
<li><details><summary>Failed: 3/59723 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer_marker.compute.default_mem.bottom_of_pipe.memory_dep.dispatch
Stack trace: Flake

FAILURE: dEQP-VK.pipeline.creation_feedback.graphics_tests.vertex_stage_fragment_stage_delayed_destroy
Stack trace: Flake

FAILURE: dEQP-VK.pipeline.creation_feedback.graphics_tests.vertex_stage_fragment_stage_no_cache
Stack trace: Flake
```
</p></details></li>

<li>Skipped: 32862/59723 (55.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchBufferOp.cpp (hunk 1 above)
Comment (trenouf) [reply lgc/patch/PatchBufferOp.cpp]: If alignment could be improved, that should be a separate change, leaving this change to just cope with the upstream API change.

**Review by trenouf — APPROVED** [review]
LGTM. I will merge.

**Pull request merged**